certificateshop-logic/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,260 @@
+        if (order.getCertificates() == null || order.getCertificates().isEmpty()) {
+            errorMessage.add(translator.toLocale("CERTIFICATES_IN_ORDER_SHOULD_BE_NOT_EMPTY"));
+            throw new MethodArgumentNotValidException(ERROR_CODE_DUPLICATE + ERROR_CODE_ORDER_NOT_VALID, errorMessage);
+        }
+        if (order.getCertificates().size() == 1) {
+            orderCertificate(order.getUser().getId(), order.getCertificates().get(0).getId());
+        }
+        User userFromDatabase = userDao.findById(order.getUser().getId()).get();
+        String newOrderName = createNewOrder(order, userFromDatabase);
+        Order orderFromDB = orderDao.findByName(newOrderName).get();
+        for (GiftCertificate certificate : order.getCertificates()) {
+
+            GiftCertificate certificateToAddToOrder = certificateDAO.findById(certificate.getId()).get();
+            orderDao.saveIdsInUserorder_certificateTable(orderFromDB.getId(), certificate.getId(),
+                    certificateToAddToOrder, certificateToAddToOrder.getPrice());
+        }
+        return orderDao.findByName(newOrderName).get();
+    }
+
+    private String createNewOrder(Order order, User user) {
+        String name = generateUniqueOrderName(user);
+        Order newOrder = new Order(0, order.getUser(), LocalDateTime.now(), name, order.getCertificates());
+        orderDao.save(newOrder);
+        return name;
+    }
+
+    private void checkUserIdAndCertificatesId(Order order, List<String> errorMessage) {
+        if (order.getUser().getId() == 0) {
+            errorMessage.add(translator.toLocale("USER_ID_SHOULD_NOT_BE_EMPTY"));
+        } else if (order.getUser().getId() < 0) {
+            errorMessage.add(
+                    String.format(translator.toLocale("SOME_ID_SHOULD_NOT_BE_LESS_THAN_ONE"), "userId"));
+        } else if (!userDao.findById(order.getUser().getId()).isPresent()) {
+            errorMessage.add(String.format(translator.toLocale(
+                    "THERE_IS_NO_A_USER_WITH_SUCH_AN_ID_IN_DATABASE"), order.getUser().getId()));
+        }
+        for (GiftCertificate certificate : order.getCertificates()) {
+            if (certificate.getId() == 0) {
+                errorMessage.add(translator.toLocale("CERTIFICATE_ID_SHOULD_NOT_BE_EMPTY"));
+            } else if (certificate.getId() < 0) {
+                errorMessage.add(String.format(
+                        translator.toLocale("SOME_ID_SHOULD_NOT_BE_LESS_THAN_ONE"), "certificateId"));
+            } else if (!certificateDAO.findById(certificate.getId()).isPresent()) {
+                errorMessage.add(String.format(translator.toLocale(
+                        "THERE_IS_NO_A_CERTIFICATE_WITH_SUCH_AN_ID_IN_DATABASE"), certificate.getId()));
+            }
+        }
+        if (!errorMessage.isEmpty()) {
+            throw new EntityNotFoundException(ERROR_CODE_ENTITY_NOT_FOUND + ERROR_CODE_ORDER_NOT_VALID, errorMessage);
+        }
+    }
```

Comment:
Divide into small private methods, because of IFs

Reply:
done